python/taichi/lang/simt/block.py

```diff
@@ -1,5 +1,19 @@
 from taichi.lang import impl
+from taichi.lang.util import taichi_scope
 
 
 def sync():
     return impl.call_internal("block_barrier", with_runtime_context=False)
+
+
+class SharedArray:
+    _is_taichi_class = True
+
+    def __init__(self, shape, dtype):
+        self.shape = shape
+        self.dtype = dtype
+        self.shared_array_proxy = impl.expr_init_shared_array(shape, dtype)
+
+    @taichi_scope
+    def _subscript(self, indices, get_ref=False):
+        return impl.make_index_expr(self.shared_array_proxy, (indices, ))
```

Comment:
@ailzhang I'm not sure if this is the best method to load an element. Furthermore, I wonder if it's possible to manually keep a pointer that refers to a partial position, like this:
```Python
for p in range(...):
    pad = ti.simt.block.SharedArray(...)
    for i in range(..):
         pad_i = pad.offset(128 + 32 * i)
         for j in range(...):
             calc(pad_i)
```

Reply:
@turbo0628 we cannot support partial position for now IIUC, as that might create aliasing issues?